### PR TITLE
An elastic response

### PR DESCRIPTION
**Pull request by AndreaNeumayr**

### CODE REVIEW

File: src/Composition/responseCalculation/elasticCollisionResponse.jl [hunk 1]
```diff
@@ -164,9 +160,9 @@ function responseCalculation(material::ElasticContactPairResponseMaterial, obj1:
     delta     = -s
 
     if c_geo != 1.0 && n_geo != 1.0
-      delta_comp = delta * sqrt(abs(delta))
+        delta_comp = delta * sqrt(abs(delta))
     else
-      delta_comp = delta
+        delta_comp = delta
```
Comment (GerhardHippmann) [inline src/Composition/responseCalculation/elasticCollisionResponse.jl:165]: Not required anymore.

File: src/Shapes/setCollisionSmoothingRadius.jl [hunk 2]
```diff
@@ -30,3 +30,37 @@ function setCollisionSmoothingRadius(shape::Beam{F}, collisionSmoothingRadius::F
...
+
+setContactSphereRadius(shape::Sphere{F}, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (false, F(shape.diameter * 0.5) )
+setContactSphereRadius(shape::Sphere{F}, contactSphereRadius::F, ::Type{F}) where F <: Modia3D.VarFloatType = (false, contactSphereRadius)
+
+
+setContactSphereRadius(shape::Ellipsoid{F}, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (false, min(shape.lengthX, shape.lengthY, shape.lengthZ)*F(0.5) )
+setContactSphereRadius(shape::Ellipsoid{F}, contactSphereRadius::F, ::Type{F}) where F <: Modia3D.VarFloatType = (false, contactSphereRadius )
+
+
+setContactSphereRadius(shape::Cylinder{F}, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (false, min(shape.diameter, shape.length)*F(0.5) )
```
Comment (GerhardHippmann) [inline src/Shapes/setCollisionSmoothingRadius.jl:45]: Isn't F() required here?

File: src/Shapes/setCollisionSmoothingRadius.jl [hunk 3]
```diff
@@ -30,3 +30,37 @@ function setCollisionSmoothingRadius(shape::Beam{F}, collisionSmoothingRadius::F
...
+
+setContactSphereRadius(shape::Box{F}, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (true, F(min(shape.lengthX, shape.lengthY, shape.lengthZ)*0.5) )
+setContactSphereRadius(shape::Box{F}, contactSphereRadius::F, ::Type{F}) where F <: Modia3D.VarFloatType = (true, contactSphereRadius )
+
+
+setContactSphereRadius(shape::Beam{F}, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (true, F(min(shape.length, shape.width, shape.thickness)*0.5) )
+setContactSphereRadius(shape::Beam{F}, contactSphereRadius::F, ::Type{F}) where F <: Modia3D.VarFloatType = (true, contactSphereRadius )
+
+
+setContactSphereRadius(shape::FileMesh, contactSphereRadius::Nothing, ::Type{F}) where F <: Modia3D.VarFloatType = (false, shape.shortestEdge )
```
Comment (GerhardHippmann) [inline src/Shapes/setCollisionSmoothingRadius.jl:65]: Should be 0.5*shortestEdge

File: src/Shapes/setCollisionSmoothingRadius.jl (hunk 2 above)
Comment (AndreaNeumayr) [reply src/Shapes/setCollisionSmoothingRadius.jl]: I changed it, it's consistent with the other functions now.

**Review by GerhardHippmann — APPROVED** [review]

**Pull request merged**